engine/src/main/java/io/zeebe/engine/processor/workflow/deployment/model/transformer/MessageTransformer.java

```diff
@@ -27,21 +29,36 @@
   public void transform(final Message element, final TransformContext context) {
 
     final String id = element.getId();
-    final ExecutableMessage executableElement = new ExecutableMessage(id);
+    final ExpressionLanguage expressionLanguage = context.getExpressionLanguage();
 
+    final ExecutableMessage executableElement = new ExecutableMessage(id);
     final ExtensionElements extensionElements = element.getExtensionElements();
 
     if (extensionElements != null) {
       final ZeebeSubscription subscription =
           extensionElements.getElementsQuery().filterByType(ZeebeSubscription.class).singleResult();
       final Expression correlationKeyExpression =
-          context.getExpressionLanguage().parseExpression(subscription.getCorrelationKey());
+          expressionLanguage.parseExpression(subscription.getCorrelationKey());
 
       executableElement.setCorrelationKeyExpression(correlationKeyExpression);
     }
 
     if (element.getName() != null) {
-      executableElement.setMessageName(BufferUtil.wrapString(element.getName()));
+      final Expression messageNameExpression =
+          expressionLanguage.parseExpression(element.getName());
+
+      executableElement.setMessageNameExpression(messageNameExpression);
+
+      if (messageNameExpression.isStatic()) {
+        final EvaluationResult messageNameResult =
+            expressionLanguage.evaluateExpression(messageNameExpression, variable -> null);
+
+        if (messageNameResult.getType() == ResultType.STRING) {
+          final String messageName = messageNameResult.getString();
+          executableElement.setMessageName(messageName);
+        }
+      }
```

Comment:
Why do we evaluate every static message name expression in the transformer?

Reply:
yeah, I thought you would stumble over this. To be honest, I grew less and less fond of it myself.

I implemented this bit in the very beginning. Back then I wanted to defer the start event handling to a different issue, like you eluded to. 

My intention was to make sure that all BPMN models that worked before, would still work. And since all BPMN models so far use static value here, they are now evaluated in the transformer. 

This was before I found `StartEventTransformer`. I think with `StartEventTransofrmer` and `CatchEventBehavior` both evaluating expressions, I might have covered all bases. But since I am unsure these are truly all cases where evaluation needs to take place, I left it in as a conservative coding measure.

Reply:
:+1: